contracts/bprotocol/scoring/BTokenScore.sol

```diff
@@ -2,21 +2,57 @@ pragma solidity 0.5.16;
 
 import { ScoringMachine } from "../../../user-rating/contracts/score/ScoringMachine.sol";
 import { IRegistry } from "../interfaces/IRegistry.sol";
+import { IComptroller } from "../interfaces/IComptroller.sol";
+import { Exponential } from "../lib/Exponential.sol";
+import { ICToken } from "../interfaces/CTokenInterfaces.sol";
+import { IBToken } from "../interfaces/IBToken.sol";
+import { IBComptroller } from "../interfaces/IBComptroller.sol";
 
-contract BTokenScore is ScoringMachine {
+contract BTokenScore is ScoringMachine, Exponential {
 
     IRegistry public registry;
+    IComptroller public comptroller;
     string private constant parent = "BTokenScore";
+    uint public expiry;
+
+    // cToken => uint (supplyMultiplier)
+    mapping(address => uint) public supplyMultiplier;
+    // cToken => uint (borrowMultiplier)
+    mapping(address => uint) public borrowMultiplier;
+
+    // cToken => LastIndex{supplyIndex, borrowIndex}
+    mapping(address => LastIndex) public lastIndex;
+    struct LastIndex {
+        uint224 supplyIndex;
+        uint224 borrowIndex;
+    }
 
     modifier onlyAvatar() {
-        address _owner = registry.ownerOf(msg.sender);
-        require(_owner != address(0), "Score: not-an-avatar");
+        require(registry.doesAvatarExist(msg.sender), "Score: not-an-avatar");
         _;
     }
 
+    constructor(
+        address[] memory cTokens,
+        uint[] memory sMultipliers,
+        uint[] memory bMultipliers
+    ) public {
+        expiry = now + (6 * 4 weeks); // 6 months
```

Comment:
this should also be an input parameter. either end date or number of seconds/days/weeks.